WordPress/gutenberg
# Deprecating `isPressed` in `Button` component

## What?
This PR starts the deprecation of `isPressed` on the `Button` component, deferring to the native `aria-pressed` attribute instead. While it does not externally change the API, internally it converts `isPressed` into `aria-pressed`, such that using the latter is equivalent to the former. As such, `aria-pressed` can be used instead of `isPressed` should a consumer so desire.

## Why?
When the `isPressed` prop is used on `Button`, `aria-pressed` is explicitly set accordingly. But sometimes `Button` is used for roles other than `button`, such as `option` and `checkbox`, where `aria-pressed` is not appropriate. Instead consumers should be able to decide for themselves the appropriate semantics.

## How?
The `isPressed` prop is marked as deprecated, and transformed into `aria-pressed`. The value of `aria-pressed` is then used instead to determine how the component should be styled.

## Testing Instructions
Unit tests have been added to confirm that the legacy `isPressed` behaviour continues, and that using `aria-pressed` directly behaves in the same way.

The only potential scope for issues is where a consumer has already set `aria-pressed` on a `Button`, but not `isPressed`, as this could affect the visual appearance due to the added `is-pressed` class.
